src/main/actions/file.js

```diff
@@ -38,7 +38,12 @@ const writeFile = (pathname, content, extension, e, callback = null) => {
   }
 }
 
-const writeMarkdownFile = (pathname, content, extension, win, e, quitAfterSave = false) => {
+const writeMarkdownFile = (pathname, content, extension, isUtf8BomEncoded, win, e, quitAfterSave = false) => {
```

Comment:
There is a question?

You can judge weather the `content` is `UTF BOM encoded`, why don't you need the variable `isUtf8BomEncoded`.

is it necessary to store it in the render/state?

Reply:
Yes and no :smile: The BOM sequence is shown as " " or hieroglyphics in the editor. So I remove it first and store encoding in `state` and later insert BOM before save. Otherwise the user could remove the BOM and you have a (space) character before the first readable character.

Reply:
809d021243d85361d810aef6ab4bcf9ab0e744d3